src/contracts/libraries/BeaconChainProofs.sol

```diff
@@ -414,4 +406,100 @@ library BeaconChainProofs {
         require(validatorPubkey.length == 48, "Input should be 48 bytes in length");
         return sha256(abi.encodePacked(validatorPubkey, bytes16(0)));
     }
+
+    /**
+     * @notice Parses a balanceRoot to get the uint64 balance of a validator.  
+     * @dev During merkleization of the beacon state balance tree, four uint64 values are treated as a single 
+     * leaf in the merkle tree. We use validatorIndex % 4 to determine which of the four uint64 values to 
+     * extract from the balanceRoot.
+     * @param balanceRoot is the combination of 4 validator balances being proven for
+     * @param validatorIndex is the index of the validator being proven for
+     * @return The validator's balance, in Gwei
+     */
+    function getBalanceAtIndex(bytes32 balanceRoot, uint40 validatorIndex) internal pure returns (uint64) {
+        uint256 bitShiftAmount = (validatorIndex % 4) * 64;
+        return 
+            Endian.fromLittleEndianUint64(bytes32((uint256(balanceRoot) << bitShiftAmount)));
+    }
+
+    /**
+     * @dev Retrieve the withdrawal timestamp
+     */
+    function getWithdrawalTimestamp(WithdrawalProof memory withdrawalProof) internal pure returns (uint64) {
+        return
+            Endian.fromLittleEndianUint64(withdrawalProof.timestampRoot);
+    }
+
+    /**
+     * @dev Converts the withdrawal's slot to an epoch
+     */
+    function getWithdrawalEpoch(WithdrawalProof memory withdrawalProof) internal pure returns (uint64) {
+        return
+            Endian.fromLittleEndianUint64(withdrawalProof.slotRoot) / SLOTS_PER_EPOCH;
+    }
+
+    /**
+     * Getters for validator fields:
+     * 0: pubkey
+     * 1: withdrawal credentials
+     * 2: effective balance
+     * 3: slashed?
+     * 4: activation elligibility epoch
+     * 5: activation epoch
+     * 6: exit epoch
```

Comment:
maybe I'm missing something, but not seeing these 4 getter functions

Reply:
Yeah, we don't have getters for all of these, because we don't need to use all of these fields. Should I update the comment to mention that?

Reply:
I really just wanted a short and dirty reference to the indices being used in the getters we do have

Reply:
OK yeah, fair enough. I think it's fine.